File: src/main/java/io/openfuture/openmessanger/assistant/gemini/GeminiResponse.java
package io.openfuture.openmessanger.assistant.gemini;

import java.util.List;

import lombok.Data;

@Data
public class GeminiResponse {
    private List<Candidate> candidates;

    @Data
    public static class Candidate {
        private Content content;
        private String finishReason;
        private int index;
        private List<SafetyRating> safetyRatings;
    }

    @Data
    public static class Content {
        private List<Part> parts;
        private String role;
    }

    @Data
    public static class Part {
        private String text;
    }

    @Data
    public static class SafetyRating {
        private String category;
        private String probability;
    }

}

File: src/main/kotlin/io/openfuture/openmessenger/assistant/gemini/GeminiResponse.kt
package io.openfuture.openmessenger.assistant.gemini

data class GeminiResponse(
    val candidates: List<Candidate>? = null
)

class Candidate(
    val content: Content? = null,
    val finishReason: String? = null,
    val index: Int = 0,
    val safetyRatings: List<SafetyRating>? = null
)

class Content(
    val parts: List<Part>? = null,
    val role: String? = null
)

class Part(
    val text: String? = null
)

class SafetyRating(
    val category: String? = null,
    val probability: String? = null
)